pkg/controllers/resources/pods/translate/translator_test.go

```diff
@@ -115,20 +124,34 @@ func TestPodAffinityTermsTranslation(t *testing.T) {
 				},
 			},
 			expectedTerm: corev1.PodAffinityTerm{
-				LabelSelector: appendToMatchLabels(&metav1.LabelSelector{
-					MatchExpressions: []metav1.LabelSelectorRequirement{
-						{
-							Key:      translate.ConvertLabelKeyWithPrefix(NamespaceLabelPrefix, longKey),
-							Operator: metav1.LabelSelectorOpNotIn,
-							Values:   []string{"bad-value"},
+				LabelSelector: translate.MergeLabelSelectors(
+					basicSelectorTranslatedWithMarker,
+					&metav1.LabelSelector{
+						MatchExpressions: []metav1.LabelSelectorRequirement{
+							{
+								Key:      translate.ConvertLabelKeyWithPrefix(NamespaceLabelPrefix, longKey),
+								Operator: metav1.LabelSelectorOpNotIn,
+								Values:   []string{"bad-value"},
+							},
 						},
 					},
-				}, translate.MarkerLabel, translate.VClusterName),
+				),
 			},
 		},
 	}
 
+	isFocused := false
+	for _, testCase := range testCases {
+		if testCase.focus {
+			isFocused = true
+			break
+		}
+	}
+
 	for _, testCase := range testCases {
+		if isFocused && !testCase.focus {
+			continue
+		}
```

Comment:
what is this for? selecting only a subset of tests to be run when developing?

Reply:
Yup, it fails at the end if any tests are focused.

Reply:
not sure I am a fan of this :D what's wrong with `go test -v` or something similar?
I think the issue here is that it's not using table driven test properly, because you'd have a proper breakdown of what tests failed with something like this 
```go

func Test(t *testing.T) {
	testCases := []struct {
		desc	string
		
	}{
		{
			desc: "",
			
		},
	}
	for _, tC := range testCases {
		t.Run(tC.desc, func(t *testing.T) {
			
		})
	}
}
```

Reply:
what do you mean? `Also, sometimes you can't get to the test case you're trying to game out until you make the previous ones pass` 
You can absolutely do that with `t.Run` 

` this code doesn't do anything till the next person has to fiddle with the tests` so it's essentially dead code no? And you can also use `go test -v -run Test/subtest` (with $ if you have subtest, subtest_something, etc so it only runs a complete match)

Reply:
Ahh, right. Didn't know you could use `-run Test/subtest` removing :)